### PR TITLE
Implemented assigning a user to a ticket and associated CLI

### PR DESCRIPTION
**Pull request by dawzee**

### CODE REVIEW

File: bug/op_set_assignee.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+func (op *SetAssigneeOperation) Validate() error {
+	if err := opBaseValidate(op, SetAssigneeOp); err != nil {
+		return err
+	}
+
+	// TODO
+	//if err := op.Assignee.Validate(); err != nil {
+	//	return errors.Wrap(err, "assignee")
+	//}
```
Comment (aleb) [inline bug/op_set_assignee.go:50]: Uncomment, or say what should be done / what's the issue.

File: commands/assign.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		}
+	}
+
+	if assigneeId == "" {
+		fmt.Printf("No users matching %s\n", user)
+		return nil
+	}
+
+	// Check the ticket is not already assigned to the new assignee
+	currentAssignee, err := backend.ResolveIdentityExcerpt(b.Snapshot().Assignee.Id())
```
[the remark this answers is not in the document]
Comment (dawzee) [reply commands/assign.go:58]: oops, yes

File: bug/operation_pack_test.go [hunk 3]
```diff
@@ -20,12 +20,24 @@ func TestOperationPackSerialize(t *testing.T) {
...
 	labelChangeOp := NewLabelChangeOperation(rene, time.Now().Unix(), []Label{"added"}, []Label{"removed"})
+	setChecklistOp := NewSetChecklistOp(rene, time.Now().Unix(), Checklist{Label: "123",
+		Title: "123 Checklist",
+		Questions: []ChecklistQuestion{
+			ChecklistQuestion{Question: "1?"},
+			ChecklistQuestion{Question: "2?"},
+			ChecklistQuestion{Question: "3?"},
+		},
+	})
+	mickey := identity.NewBare("Mickey Mouse", "mm@disney.com")
```
[the remark this answers is not in the document]
Comment (dawzee) [reply bug/operation_pack_test.go:31]: They work together in another universe =)

File: cache/bug_cache.go [hunk 4]
```diff
@@ -188,13 +188,26 @@ func (c *BugCache) CloseRaw(author *IdentityCache, unixTime int64, metadata map[
...
+
+func (c *BugCache) SetAssigneeRaw(author *IdentityCache, unixTime int64, metadata map[string]string, assignee *IdentityCache) (*bug.SetAssigneeOperation, error) {
+	op, err := bug.SetAssignee(c.bug, author.Identity, unixTime, assignee.Identity)
+	if err != nil {
+		return nil, err
+	}
+
+	for key, value := range metadata {
+		op.SetMetadata(key, value)
+	}
```
[the remark this answers is not in the document]
Comment (dawzee) [reply cache/bug_cache.go:208]: added to be consistent with all the other XxxxRaw funcs

File: bug/op_set_assignee.go (hunk 1 above)
Comment (dawzee) [reply bug/op_set_assignee.go:50]: I've just removed it. Thought I'd need to do some identity validation at this point but realised that it doesn't make sense here.

**Comment by aleb** [comment]
LGTM

**Pull request merged**